src/main/java/frc/robot/subsystems/ShooterFlyWheel.java

```diff
@@ -7,18 +7,127 @@
+import static frc.robot.Constants.ShooterConstants.leftShooterNeoPort;
+import static frc.robot.Constants.ShooterConstants.rightShooterNeoPort;
+import static frc.robot.Constants.ShooterConstants.shooterD;
+import static frc.robot.Constants.ShooterConstants.shooterF;
+import static frc.robot.Constants.ShooterConstants.shooterI;
+import static frc.robot.Constants.ShooterConstants.shooterP;
+
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANPIDController;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.revrobotics.ControlType;
+
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
 public class ShooterFlyWheel extends SubsystemBase {
-  /**
-   * Creates a new ShooterFlyWheel.
+
+  private CANSparkMax leftShooterNeo;
+  private CANSparkMax rightShooterNeo;
+
+  private CANPIDController PIDController;
+
+  private CANEncoder flyWheelEncoder;
+
+  /*These thresholds are used by commands to decide how close the flywhell's velocity needs to be to
+  its setpoint before a ball is fed.
    */
+  public final double accurateShootVelocityThreshhol = 0.95;
+  public final double fastShootVelocityThreshold = 0.8;
+
+  //target height and shooter height in meters
+  public final double targetHeight = 5;
+  public final double shooterHeight = 4;
+
   public ShooterFlyWheel() {
 
+    leftShooterNeo = new CANSparkMax(leftShooterNeoPort, MotorType.kBrushless);
+    rightShooterNeo = new CANSparkMax(rightShooterNeoPort, MotorType.kBrushless);
+
+    leftShooterNeo.follow(rightShooterNeo, true);
+
+    leftShooterNeo.setIdleMode(IdleMode.kCoast);
+    rightShooterNeo.setIdleMode(IdleMode.kCoast);
+
+    flyWheelEncoder = rightShooterNeo.getEncoder();
+    PIDController = rightShooterNeo.getPIDController();
+    PIDController.setFeedbackDevice(flyWheelEncoder);
+
+    updatePIDConstants();
+    stop();
   }
 
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
   }
-}
+
+  /**
+   * Updates the constants for the PID using constants.
+   */
+  public void updatePIDConstants() {
+    PIDController.setOutputRange(0, 1);
+    PIDController.setP(shooterP);
+    PIDController.setI(shooterI);
+    PIDController.setD(shooterD);
+    PIDController.setFF(shooterF);
+  }
+
+  /**
+   * Sets the PID target from surface velocity.
+   *
+   * @param surfaceVelocity the speed that the surface of the ball goes in meters per second.
+   */
+  public void setFlyWheelSpeed(double surfaceVelocity) {
+    PIDController.setReference(surfaceVelocityToRPM(surfaceVelocity), ControlType.kVelocity);
+  }
+
+  /**
+   * Calculates the surface velocity from neo RPM.
+   *
+   * @retrun calculated surface velocity.
+   */
+  public double getFlyWheelSpeed() {
+    return rpmToSurfaceVelocity(rightShooterNeo.getEncoder().getVelocity());
+  }
+
+  /**
+   * Sets the PID target to zero, letting it coast to a stop.
```

Comment:
Is this the right wording?  Setting the setpoint to 0 would make the motor run to the target of 0 which is not what we want

Reply:
idek man